plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionData.java

```diff
@@ -137,6 +137,7 @@ public static Object getValue(JsonNode partitionValue, Type type)
                 return partitionValue.asInt();
             case LONG:
             case TIMESTAMP:
+            case TIMESTAMP_NANO:
```

Comment:
nano will be mapped to long timestamp, so this won't be there.

keep as not implemented and add a todo + link to https://github.com/trinodb/trino/issues/19753